C7GameData/AIData/TilePath.cs

```diff
@@ -52,9 +53,29 @@ public int PathCost(Tile from, float perTurnMovePoints, float remainingMovementP
 
 				from = tile;
 			}
+
+			// Special case: if we consumed part of our movement points (such as by
+			// walking along a road, consuming 1/3 of a point), return a cost of 1.
+			// This doesn't apply when railroads are used, since that wouldn't change
+			// movementPoints.remaining. If it wasn't for railroads we could just
+			// return Max(turns, 1).
+			if (turns == 0 && movementPoints.remaining != remainingMovementPoints) {
+				return 1;
+			}
```

Comment:
True, but it still needs to round up if any movement points are consumed on the final turn. For example if I goto 4 or 5 tiles along a road it shows 1 even though the movement won't complete until the second turn. The general case for this final check would be something like `if (movementPoints.remaining < (turns == 0 ? remainingMovementPoints : perTurnMovePoints) ) ++turns`

Reply:
Ah, good catch. I hadn't tested 4 moves along a road. Your suggestion worked perfectly, thanks.